main/GarnetServer/Program.cs

```diff
@@ -16,82 +12,17 @@ namespace Garnet
     /// </summary>
     class Program
     {
-        private static readonly string CustomRespCommandInfoJsonFileName = "CustomRespCommandsInfo.json";
-
         static void Main(string[] args)
         {
-            try
-            {
-                using var server = new GarnetServer(args);
-
-                // Optional: register custom extensions
-                if (!TryRegisterExtensions(server))
-                {
-                    Console.WriteLine("Unable to register server extensions.");
-                    return;
-                }
-
-                // Start the server
-                server.Start();
-
-                Thread.Sleep(Timeout.Infinite);
-            }
-            catch (Exception ex)
+            var builder = Host.CreateEmptyApplicationBuilder(null);
+            builder.Services.AddHostedService(sp => new GarnetService(args));
+            builder.Services.AddWindowsService(options =>
```

Comment:
As this says Windows service, just checking how does this work on Linux and other platforms?

Reply:
It has no effect on other OS.

Reply:
It's also no-op for if the process is not launched as service, see https://github.com/microsoft/garnet/pull/394#discussion_r1605099149.

There is similar extension for Systemd integration.